hyp3_testing/opera_compare.py

```diff
@@ -0,0 +1,325 @@
+                f'Values for key {str_key} do not match ({val_1} | {val_2})'
+            )
+            return
+        assert np.array_equal(val_1, val_2)
+        return
+
+    if len(shape_val_1) >= 2:
+        assert np.allclose(val_1, val_2, **ALL_CLOSE_ARGS), f'Values for key {str_key} do not match ({val_1} | {val_2})'
+        return
+
+    # Unexpected failure to compare `val_1` and `val_2`
+    raise ValueError(
+        f'Failed to compare the element: {str_message_data_location}'
+        f'dataset shape in the 1st HDF5: {shape_val_1}'
+        f'dataset shape in the 2nd HDF5: {shape_val_2}'
+    )
+
+
+def compare_rtc_hdf5_files(file_1: Path, file_2: Path):
+    """
+    Compare the two RTC products (in HDF5) if they are equivalent
+    within acceptable difference
+
+    Args:
+        file_1: Path to the first HDF5 file
+        file_2: Path to the second HDF5 file
+    """
+    assert file_1.exists()
+    assert file_2.exists()
+    with h5py.File(str(file_1), 'r') as hdf5_in_1, h5py.File(str(file_2), 'r') as hdf5_in_2:
+        list_dataset_1, list_attrs_1 = get_list_dataset_attrs_keys(hdf5_in_1)
+        set_dataset_1 = set(list_dataset_1)
+        set_attrs_1 = set(list_attrs_1)
+
+        list_dataset_2, list_attrs_2 = get_list_dataset_attrs_keys(hdf5_in_2)
+        set_dataset_2 = set(list_dataset_2)
+        set_attrs_2 = set(list_attrs_2)
+
+        intersection_set_dataset = set_dataset_1.intersection(set_dataset_2)
+        assert len(intersection_set_dataset) == len(set_dataset_1)
+        assert len(intersection_set_dataset) == len(set_dataset_2)
+
+        intersection_set_attrs = set_attrs_1.intersection(set_attrs_2)
+        assert len(intersection_set_attrs) == len(set_attrs_1)
+        assert len(intersection_set_attrs) == len(set_attrs_2)
+
+        # Proceed with checking the values in dataset,
+        # regardless of the agreement of their structure.
+        for key_dataset in intersection_set_dataset:
+            compare_hdf5_elements(hdf5_in_1, hdf5_in_2, key_dataset, is_attr=False)
+
+        # Proceed with checking the values in attributes,
+        # regardless of the agreement of their structure.
+        for key_attr in intersection_set_attrs:
+            compare_hdf5_elements(hdf5_in_1, hdf5_in_2, key_attr, is_attr=True)
+
+
+def elements_equal(root, e1, e2):
+    nested = ['{http://earthdata.nasa.gov/schema/eos}value']
+    assert e1.tag == e2.tag, f'Tag mismatch at {e1.tag}: {e1.tag} != {e2.tag}'
+    assert len(e1) == len(e2), f'Children count mismatch at {e1.tag}: {len(e1)} != {len(e2)}'
+    assert e1.attrib == e2.attrib, f'Attribute mismatch at {e1.tag}: {e1.attrib} != {e2.attrib}'
+    assert e1.nsmap == e2.nsmap, f'Namespace mismatch at {e1.tag}: {e1.nsmap} != {e2.nsmap}'
+
+    full_path = root.getroottree().getpath(e1)
+    if full_path not in LIST_EXCLUDE_COMPARISON_XML:
+        # if not (e1.text or '').strip() == (e2.text or '').strip():
```

Comment:
Does this commented-out `# if not (e1.text or '').strip() == (e2.text or '').strip():` still need to be here?

Reply:
Nope, thanks!